wallet/src/de/schildbach/wallet/ui/transactions/TransactionResultViewBinder.kt

```diff
@@ -189,63 +169,135 @@ class TransactionResultViewBinder(
         } else {
             fiatValue.isVisible = false
         }
+    }
+
+    private fun updateStatus(fromConfidence: Boolean = false) {
+        val errorStatus = resourceMapper.getErrorName(transaction)
+        isError = errorStatus != -1
+        val errorStatusStr = if (isError) {
+            ctx.getString(errorStatus)
+        } else {
+            ""
+        }
+
+        setTransactionDirection(errorStatusStr)
+
+        if (!fromConfidence || isError) {
+            // If it's a confidence update, not need to set the send/receive icons again.
+            // Some hosts are replacing those with custom animated ones.
+            updateIcon()
+        }
+    }
+
+    fun setTransactionMetadata(transactionMetadata: TransactionMetadata) {
+        val strResource = if (transactionMetadata.taxCategory != null) {
+            taxCategoryNames[transactionMetadata.taxCategory!!]
+        } else {
+            taxCategoryNames[transactionMetadata.defaultTaxCategory]
+        }
+        taxCategory.text = containerView.resources.getString(strResource!!)
+
+        if (transactionMetadata.service == ServiceName.DashDirect) {
+            iconRes = R.drawable.ic_gift_card_tx
+        }
+
+        updateIcon()
+    }
+
+    fun setTransactionIcon(bitmap: Bitmap) {
+        iconBitmap = bitmap
+        updateIcon()
+    }
+
+    fun setTransactionIcon(@DrawableRes drawableRes: Int) {
+        iconRes = drawableRes
+        updateIcon()
+    }
+
+    private fun updateIcon() {
+        val iconRes = if (isError) {
+            R.drawable.ic_transaction_failed
+        } else if (iconRes != null) {
+            iconRes!!
+        } else if (transaction.getValue(wallet).signum() >= 0) {
+            R.drawable.ic_transaction_received
+        } else if (transaction.isEntirelySelf(wallet)) {
+            R.drawable.ic_shuffle
+        } else {
+            R.drawable.ic_transaction_sent
+        }
```

Comment:
Some of the changes in this class are lifted from DashPay where we already have custom icons for profile pictures; however additional refactoring has been made to simplify and cleanup the code. This will cause some conflicts.